virtcontainers/container.go

```diff
@@ -405,6 +405,14 @@ func (c *Container) unmountHostMounts() error {
 					"error":     err,
 				}).Warn("Could not umount")
 				return err
+			} else if err := os.RemoveAll(m.HostPath); err != nil {
+				// since the mounts related to the shared dir is umounted
+				// we need to remove the host path to avoid resource remaining
+				c.Logger().WithFields(logrus.Fields{
```

Comment:
This log call essentially simulates a call to `WithError()` and although it is simpler than having to call `WithError()`, I think it's safer to explicitly call `WithError()` to protect against it changing its behaviour at some future point.

Since this is based on the log call code above, I'd do something like:

```
logger := c.Logger().WithField("host-path", m.HostPath)

if err := syscall.Unmount(m.HostPath, 0); err != nil {
    logger.WithError(err).Warn("could not unmount")
} else if err := os.RemoveAll(m.HostPath); err != nil {
    logger.WithError(err).Warn("could not remove path")
}
```

Reply:
Going to change it now~